benchmark/utils/formats.hpp

```diff
@@ -234,6 +237,21 @@ auto create_matrix_type_with_gpu_strategy()
 const std::map<std::string, std::function<std::unique_ptr<gko::LinOp>(
                                 std::shared_ptr<const gko::Executor>)>>
     matrix_type_factory{
+        {"bccoo", create_matrix_type<bccoo>()},
+        {"bccoo-elm", create_matrix_type<bccoo>(0, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-32", create_matrix_type<bccoo>(32, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-64", create_matrix_type<bccoo>(64, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-128", create_matrix_type<bccoo>(128, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-256", create_matrix_type<bccoo>(256, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-512", create_matrix_type<bccoo>(512, gko::matrix::bccoo::compression::element)},
+        {"bccoo-elm-1024", create_matrix_type<bccoo>(1024, gko::matrix::bccoo::compression::element)},
+        {"bccoo-blk", create_matrix_type<bccoo>(0, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-32", create_matrix_type<bccoo>(32, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-64", create_matrix_type<bccoo>(64, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-128", create_matrix_type<bccoo>(128, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-256", create_matrix_type<bccoo>(256, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-512", create_matrix_type<bccoo>(512, gko::matrix::bccoo::compression::block)},
+        {"bccoo-blk-1024", create_matrix_type<bccoo>(1024, gko::matrix::bccoo::compression::block)},
```

Comment:
should we keep these option in the end? or you will choose some option in the end

Reply:
All these formats are required to check the implementations and to find the best format for each architecture